lib/ansible/plugins/lookup/password.py

```diff
@@ -276,10 +279,32 @@ def run(self, terms, variables, **kwargs):
             b_path = to_bytes(path, errors='surrogate_or_strict')
             chars = _gen_candidate_chars(params['chars'])
 
+            content = None
             changed = False
-            content = _read_password_file(b_path)
-
-            if content is None or b_path == to_bytes('/dev/null'):
+            writer_process = False
+            if not os.path.exists(b_path) and b_path != to_bytes('/dev/null'):
+                try:
+                    b_pathdir = os.path.dirname(b_path)
+                    makedirs_safe(b_pathdir, mode=0o700)
+                    lockfile = os.path.join(b_pathdir, LOCKFILE_NAME)
+                    fd = os.open(lockfile, os.O_CREAT | os.O_EXCL)
+                    writer_process = True
+                    os.close(fd)
+                except OSError as e:
+                    if e.strerror == 'File exists':
+                        timeout = 0
+                        while os.path.exists(lockfile):
+                            time.sleep(0.1)
+                            timeout += 1
+                            if timeout > 20:
+                                raise AnsibleError("Password lookup cannot get the lock in 2 seconds, abort..."
+                                                   "This may caused by un-removed lockfile, you can manually remove it at %s and try again" % lockfile)
```

Comment:
I still think the message could remind where is the file (like, on controller, vs on remote system). It should be obvious for people, yes, but not every user realize that lookup are on controller side.

Reply:
Make sense, I will fix this.